.travis-build.sh

```diff
@@ -32,6 +32,23 @@ if  [ $TRAVIS_SECURE_ENV_VARS = false ] &&
     	exit 1
 fi
 
+if [ "${FORMATTING}" = "space-test" ]; then
+	cd "${TRAVIS_BUILD_DIR}"
+	echo "Testing for trailing and all sorts of broken white spaces"
+	echo "TRAVIS_COMMIT_RANGE: ${TRAVIS_COMMIT_RANGE}"
+	first_commit=`echo ${TRAVIS_COMMIT_RANGE} | sed -r 's/\..*//'`
+	git reset -q "${first_commit}"
+##	git reset -q HEAD~1
```

Comment:
@d12frosted  looks like this line was a bit too much.  "in depth"  :smile: 

At the same time it's a pretty neat idea if you think about it. When someone opens a PR CI will fail if someone committed malformed code before this person so the PR author will have to fix it. Hmmmhmmm :smiling_imp: